From 40333387723e64aa1889ea2f6a57c9a0da05d4c5 Mon Sep 17 00:00:00 2001
Message-Id: <40333387723e64aa1889ea2f6a57c9a0da05d4c5.1424979244.git.feitong.yi@intel.com>
In-Reply-To: <d8466aca83b11ab8b0870e775536306782d73e9f.1424979244.git.feitong.yi@intel.com>
References: <d8466aca83b11ab8b0870e775536306782d73e9f.1424979244.git.feitong.yi@intel.com>
From: Ramalingam C <ramalingam.c@intel.com>
Date: Mon, 2 Feb 2015 15:46:46 +0530
Subject: [PATCH 13/28] MUST_REBASE [VPG]: video/adf/intel: Interfacing the
 Idleness DRRS to the system

Initialization call intel_drrs_init() is made for all interfaces.
And to disable and reenable the Idleness detection, corresponding calls are
placed on the flip and modeset paths.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-5610
Change-Id: Ic4c58a0da96123badab95df36459f8320896792c
Signed-off-by: Ramalingam C <ramalingam.c@intel.com>
---
 .../video/adf/intel/core/common/dsi/gen_dsi_pipe.c |    9 +++++++++
 drivers/video/adf/intel/intel_adf.c                |    8 +++++++-
 2 files changed, 16 insertions(+), 1 deletion(-)

diff --git a/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c b/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
index 158892f..23c228a 100644
--- a/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
+++ b/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
@@ -22,6 +22,7 @@
 #include <core/common/dsi/dsi_pipe.h>
 #include <core/common/dsi/dsi_config.h>
 #include <core/common/intel_gen_backlight.h>
+#include <core/common/intel_drrs.h>
 /* FIXME: remove this once gpio calls are abstracted */
 #include <core/vlv/vlv_dc_config.h>
 #include <core/vlv/vlv_dc_regs.h>
@@ -352,6 +353,7 @@ static int dsi_modeset(struct intel_pipe *pipe,
 		struct drm_mode_modeinfo *mode)
 {
 	struct dsi_pipe *dsi_pipe = to_dsi_pipe(pipe);
+	struct intel_pipeline *pipeline = dsi_pipe->base.pipeline;
 	struct dsi_config *config = &dsi_pipe->config;
 	int err = 0;
 
@@ -375,6 +377,8 @@ static int dsi_modeset(struct intel_pipe *pipe,
 		return err;
 	}
 
+	intel_disable_idleness_drrs(pipeline);
+
 	mutex_lock(&config->ctx_lock);
 
 	/* Avoiding i915 enter into DPMS */
@@ -383,6 +387,7 @@ static int dsi_modeset(struct intel_pipe *pipe,
 	dsi_display_on(pipe, mode);
 	mutex_unlock(&config->ctx_lock);
 
+	intel_restart_idleness_drrs(pipeline);
 	return err;
 }
 
@@ -441,6 +446,8 @@ static void dsi_on_post(struct intel_pipe *pipe)
 		dsi_pipe->ops.on_post(dsi_pipe);
 
 	vlv_pm_on_post(intel_config, pipe);
+
+	intel_restart_idleness_drrs(pipeline);
 }
 
 static void dsi_pre_validate(struct intel_pipe *pipe,
@@ -461,6 +468,8 @@ static void dsi_pre_post(struct intel_pipe *pipe)
 	struct vlv_pipeline *vlv_pipeline = to_vlv_pipeline(pipeline);
 	struct intel_dc_config *intel_config = &vlv_pipeline->config->base;
 
+	intel_disable_idleness_drrs(pipeline);
+
 	vlv_pm_pre_post(intel_config, pipeline, pipe);
 }
 
diff --git a/drivers/video/adf/intel/intel_adf.c b/drivers/video/adf/intel/intel_adf.c
index aa21e8c..632c9c3 100644
--- a/drivers/video/adf/intel/intel_adf.c
+++ b/drivers/video/adf/intel/intel_adf.c
@@ -13,6 +13,7 @@
  */
 #include <drm/i915_adf.h>
 #include <intel_adf.h>
+#include <core/common/intel_drrs.h>
 
 const struct intel_adf_context *g_adf_context;
 
@@ -308,7 +309,7 @@ struct intel_adf_context *intel_adf_context_create(struct pci_dev *pdev)
 	struct intel_dc_config *config;
 	struct intel_adf_interface *intfs;
 	struct intel_adf_overlay_engine *engs;
-	int n_intfs, n_engs;
+	int n_intfs, n_engs, cnt;
 #if defined(CONFIG_ADF_FBDEV) && defined(CONFIG_ADF_INTEL_FBDEV)
 	struct adf_fbdev *fbdevs;
 #endif
@@ -415,6 +416,11 @@ struct intel_adf_context *intel_adf_context_create(struct pci_dev *pdev)
 		goto err;
 	}
 
+	for (cnt = 0; cnt < n_intfs; cnt++) {
+		if (intfs[cnt].pipe->type == INTEL_PIPE_DSI)
+			intel_drrs_init(intfs[cnt].pipe->pipeline);
+	}
+
 	g_adf_context = ctx;
 
 #if defined(CONFIG_ADF_FBDEV) && defined(CONFIG_ADF_INTEL_FBDEV)
-- 
1.7.9.5

